frontend/src/pages/functional_components/RecurringRunDetailsV2FC.tsx

```diff
@@ -33,75 +32,120 @@ import { triggerDisplayString } from 'src/lib/TriggerUtils';
+  const [getRecurringRunErrMsg, setGetRecurringRunErrMsg] = useState<string>('');
+  const [getExperimentErrMsg, setGetExperimentErrMsg] = useState<string>('');
+
+  // Related to Api Response
+  const [experimentName, setExperimentName] = useState<string>();
+  const [experimentIdFromApi, setExperimentIdFromApi] = useState<string>();
+  const [recurringRunName, setRecurringRunName] = useState<string>();
+  const [recurringRunIdFromApi, setRecurringRunIdFromApi] = useState<string>();
+  const [recurringRunStatus, setRecurringRunStatus] = useState<V2beta1RecurringRunStatus>();
+
   const recurringRunId = props.match.params[RouteParams.recurringRunId];
   const Refresh = () => setRefresh(refreshed => !refreshed);
 
-  const { data: recurringRun, refetch: refetchRecurringRun } = useQuery<
-    V2beta1RecurringRun | undefined,
-    Error
-  >(
-    ['recurringRun'],
+  const {
+    data: recurringRun,
+    error: getRecurringRunError,
+    refetch: refetchRecurringRun,
+  } = useQuery<V2beta1RecurringRun, Error>(
+    ['recurringRun', recurringRunId],
     async () => {
-      try {
-        return await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
-      } catch (err) {
-        const errorMessage = await errorToMessage(err);
-        updateBanner({
-          additionalInfo: errorMessage ? errorMessage : undefined,
-          message:
-            `Error: failed to retrieve recurring run: ${recurringRunId}.` +
-            (errorMessage ? ' Click Details for more information.' : ''),
-          mode: 'error',
-        });
-        return;
-      }
+      const recurringRun = await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
+      setRecurringRunName(recurringRun.display_name);
+      setRecurringRunStatus(recurringRun.status);
+      setRecurringRunIdFromApi(recurringRun.recurring_run_id);
+
+      return recurringRun;
     },
     { enabled: !!recurringRunId, staleTime: 0, cacheTime: 0 },
   );
 
   const experimentId = recurringRun?.experiment_id!;
-  const { data: experiment } = useQuery<V2beta1Experiment | undefined, Error>(
+  const { error: getExperimentError } = useQuery<void, Error>(
     ['experiment'],
     async () => {
-      try {
-        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
-      } catch (err) {
-        const errorMessage = await errorToMessage(err);
-        updateBanner({
-          additionalInfo: errorMessage ? errorMessage : undefined,
-          message:
-            `Error: failed to retrieve this recurring run's experiment.` +
-            (errorMessage ? ' Click Details for more information.' : ''),
-          mode: 'warning',
-        });
-        return;
-      }
+      const experiment = await Apis.experimentServiceApiV2.getExperiment(experimentId);
+      setExperimentName(experiment.display_name);
+      setExperimentIdFromApi(experiment.experiment_id);
+      // The logic related to experiment is done in useQuery, so no need to return.
```

Comment:
This is getting better, but I would like to call out that we should keep the `query logic` simple, which means it usually should do only the data fetching job, and return the query result. We should still have a useEffect logic to update corresponding `useState` field, for example:
```
const { data: experiment, error: getExperimentError } = useQuery<V2beta1Experiment, Error>(
    ['experiment'],
    async () => {
        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
        },
        { enabled: !!experimentId, staleTime: 0 },
    );

  useEffect(() => {
      setExperimentName(experiment.display_name);
      setExperimentIdFromApi(experiment.experiment_id);
  }, [experiment]);
```